### PR TITLE
feat: display connectivity status (AR-1947)

### PR DESCRIPTION
**Pull request by vitorhugods**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1947" title="AR-1947" target="_blank">StoryAR-1947</a>  Offline indication (that self is offline)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

The current online/offline indication is not clear and was put in place basically as a debugging tool a few months ago.

### Solutions

> **Note**: Most of the diff changes are splitting the Composables into smaller functions (so in the future the diffs get smaller), and tests.

1. Remove the current Sync Bar
2. Implement the definitive Online/Offline indication, that stays in the same place as the `Ongoing Call` indication.

For this, I've adapted the `Ongoing Call` thing so it's a single thing called `ConnectivityUIState`, and instead of nullable stuff, it holds a sealed interface:
- `None`: No indication, we're online.
- `WaitingConnection`: Waiting for Network
- `Connecting`: Dear User, we're trying, almost there!
- `EstablishedCall`: Info about the current call

So, `CommonTopAppBarViewModel` will observe connectivity status, the current screen and the current active call (if it exists) to select one of the states above.

#### Test Coverage

- [X] I have added automated test to this contribution

#### How to Test

Drop your internet connection and re-establish it.

#### No connection -> Connecting -> Connected
https://user-images.githubusercontent.com/9389043/185031993-b137d5d7-def1-4285-8e2e-ae868c06ea0a.mp4

#### Ongoing Call -> Call is dropped by the other user
https://user-images.githubusercontent.com/9389043/185031986-a2943104-fbd0-4197-9a08-5027b4e17944.mp4

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-861/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4174254)(4174254)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4174254)(4174254)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/common/topappbar/CommonTopAppBarViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+@HiltViewModel
+class CommonTopAppBarViewModel @Inject constructor(
+    private val navigationManager: NavigationManager,
+    private val establishedCalls: ObserveEstablishedCallsUseCase,
+    private val currentScreenManager: CurrentScreenManager,
+    private val observeSyncState: ObserveSyncStateUseCase,
+) : CommonTopAppBarBaseViewModel() {
+
+    var callState by mutableStateOf(ConnectivityUIState(ConnectivityUIState.Info.None))
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/common/topappbar/CommonTopAppBarViewModel.kt:34]:
```suggestion
    var connectivityState by mutableStateOf(ConnectivityUIState(ConnectivityUIState.Info.None))
```

**Review by alexandreferris — APPROVED** [review]
Amazing work 👏🏻 loved the animation now 😍 

I just have a super simple suggestion on the name change for the ui state, other than that 👏🏻 👏🏻

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-861/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4176167)(4176167)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4176167)(4176167)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-861/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4176480)(4176480)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4176480)(4176480)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-861/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4176917)(4176917)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4176917)(4176917)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-861/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4177097)(4177097)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4177097)(4177097)-dev-release.apk)

**Pull request merged**